### PR TITLE
Fix LR reduce on plateau interaction with base LR decay

### LINKED ISSUE #3049: Learning rate reduction not functional
**Describe the bug**
Despite the respective settings in the `trainer` section, the learning rate does not decay when a plateau is reached.

**To Reproduce**

I am using ludwig for image classification with the following .yaml
```
input_features:
- name: image_path
  type: image
  encoder: resnet
  resnet_size: 101
  num_filters: 16
  kernel_size: 3
  conv_stride: 1
  batch_norm_momentum: 0.9
  batch_norm_epsilon: 0.001
  num_fc_layers: 1
  output_size: 256
  use_bias: true
  bias_initializer: zeros
  norm: null
  norm_params: null
  activation: relu
  dropout: 0

output_features:
 - name: class
   type: category

trainer:
  epochs: 50
  batch_size: 16
  early_stop: 8
  learning_rate: 0.001
  reduce_learning_rate_on_plateau: 4
  reduce_learning_rate_on_plateau_patience: 4
  reduce_learning_rate_on_plateau_rate: 0.5
```
I train the model distributed with horovod in a ludwig stock docker container:
```
horovodrun -np 3 ludwig train --config model_definition.yaml --dataset annotations.csv --experiment_name resnet101
```

In this particular case the learning rate the model describes is even `0.003`. An Excerpt from `training_progress.json`:

```
    "best_increase_batch_size_eval_metric": Infinity,
    "checkpoint_number": 34,
    "epoch": 34,
    "last_improvement_steps": 6512,
    "last_increase_batch_size": 0,
    "last_increase_batch_size_eval_metric_improvement": 0,
    "last_increase_batch_size_steps": 0,
    "last_learning_rate_reduction": 0,
    "last_learning_rate_reduction_steps": 0,
    "learning_rate": 0.003,
    "num_increases_batch_size": 0,
    "num_reductions_learning_rate": 0,
    "steps": 27676,
```

**Expected behavior**
The learning rate should be reduced after 4 epochs on no further loss improvement/

**Environment (please complete the following information):**

- OS: Linux Docker environment
- Version: Docker ludwigai/ludwig-ray-gpu:master
- Python version 3.8.5
- Ludwig version 0.7.beta

**Additional context**
this does not seem to be ludwig version specific. I experienced a similar behavior with ludwig 0.5.5

Thanks so much for some feedback and keep up the great work.

### PR DESCRIPTION
**Pull request by tgaddair**
Fixes #3049.

### CODE REVIEW

File: ludwig/modules/lr_scheduler.py [hunk 1]
```diff
@@ -83,8 +106,18 @@ def eval_step(self, progress_tracker: ProgressTracker, validation_field: str):
...
         self._eval_scheduler.step(last_metric_value)
 
+        num_reductions = self._eval_scheduler.num_reduce_lr
+        if num_reductions > prev_num_reductions:
+            # LR reduction -> update progress tracker
+            progress_tracker.last_learning_rate_reduction_steps = progress_tracker.steps
+            progress_tracker.last_learning_rate_reduction = 0
+            progress_tracker.num_reductions_learning_rate += 1
+        else:
+            progress_tracker.steps - progress_tracker.last_learning_rate_reduction_steps
```
Comment (justinxzhao) [inline ludwig/modules/lr_scheduler.py:119]: Should this be: `progress_tracker.last_learning_rate_reduction_steps = progress_tracker.steps - progress_tracker.last_learning_rate_reduction_steps`?

**Review by justinxzhao — APPROVED** [review]
LGTM, save for one comment.

File: ludwig/modules/lr_scheduler.py (hunk 1 above)
Comment (tgaddair) [reply ludwig/modules/lr_scheduler.py:119]: Ah, good catch!
